java/java.mx.project/src/org/netbeans/modules/java/mx/project/SuiteSources.java

```diff
@@ -0,0 +1,1195 @@
+            Compliance.Provider {
+        private final String mxName;
+        private final MxProject mxPrj;
+        private final FileObject srcDir;
+        private final FileObject srcGenDir;
+        private final FileObject binDir;
+        private final String name;
+        private final String displayName;
+        private final Compliance compliance;
+        private ClassPath sourceCP;
+        private ClassPath cp;
+        private ClassPath processorPath;
+        private Collection<Dep> allDeps;
+
+        Group(String mxName, MxProject mxPrj, FileObject srcDir, FileObject srcGenDir, FileObject binDir, String name, String displayName) {
+            this.mxName = mxName;
+            this.mxPrj = mxPrj;
+            this.srcDir = srcDir;
+            this.srcGenDir = srcGenDir;
+            this.binDir = binDir;
+            this.name = name;
+            this.displayName = displayName;
+            this.compliance = Compliance.parse(mxPrj.javaCompliance());
+        }
+
+        @Override
+        public FileObject getRootFolder() {
+            return srcDir;
+        }
+
+        @Override
+        public String getName() {
+            return name;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return displayName;
+        }
+
+        @Override
+        public Icon getIcon(boolean opened) {
+            return null;
+        }
+
+        @Override
+        public Compliance getCompliance() {
+            return compliance;
+        }
+
+        @Override
+        public boolean contains(FileObject file) {
+            if (file == srcDir || file == srcGenDir || FileUtil.isParentOf(srcDir, file) || (srcGenDir != null && FileUtil.isParentOf(srcGenDir, file))) {
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public void addPropertyChangeListener(PropertyChangeListener l) {
+        }
+
+        @Override
+        public void removePropertyChangeListener(PropertyChangeListener l) {
+        }
+
+        @Override
+        public String toString() {
+            return "SuiteSources.Group[name=" + name + ",rootFolder=" + srcDir + "]"; // NOI18N
+        }
+
+        ClassPath getSourceCP() {
+            computeTransitiveDeps();
+            return sourceCP;
+        }
+
+        ClassPath getCP() {
+            computeTransitiveDeps();
+            return cp;
+        }
+
+        @Override
+        public Collection<String> depNames() {
+            return mxPrj.dependencies();
+        }
+
+        @Override
+        public void setAllDeps(Collection<Dep> set) {
+            allDeps = set;
+        }
+
+        @Override
+        public Collection<Dep> allDeps() {
+            return allDeps;
+        }
+
+        private void computeClassPath(Map<String, Dep> transDeps) {
+            for (Dep d : transDeps.values()) {
+                d.owner().computeTransitiveDeps();
```

Comment:
`processTransDeps` computes transitive deps on `d.owner()`, isn't it computed 2nd time here for all deps not just those used in classpath construction ?

Reply:
The transitive deps are processed just once. Let's make it clearer by better naming 5b589ee5930f